rust/src/writer.rs

```diff
@@ -0,0 +1,254 @@
+//! The writer module contains the DeltaWriter and DeltaWriterError definitions and provides a
+//! higher level API for performing Delta transaction writes to a given Delta Table.
+//!
+//! Unlike the transaction API on DeltaTable, this higher level writer will also write out the
+//! parquet files
+
+use crate::action::Txn;
+use crate::{DeltaTableError, DeltaTransactionError};
+use arrow::record_batch::RecordBatch;
+use parquet::arrow::ArrowWriter;
+use parquet::basic::Compression;
+use parquet::file::properties::WriterProperties;
+use parquet::file::writer::InMemoryWriteableCursor;
+use serde_json::Value;
+use std::collections::HashMap;
+use std::convert::TryFrom;
+use std::sync::Arc;
+
+/**
+ * BufferedJSONWriter allows for buffering serde_json::Value rows before flushing to parquet files
+ * and a Delta transaction
+ */
+pub struct BufferedJSONWriter {
+    table: crate::DeltaTable,
+    buffer: HashMap<WriterPartition, Vec<Value>>,
+    schema: arrow::datatypes::SchemaRef,
+    partitions: Vec<String>,
+    txns: Vec<Txn>,
+}
+
+impl BufferedJSONWriter {
+    /**
+     * Attempt to construct the BufferedJSONWriter, will fail if the table's metadata is not
+     * present
+     */
```

Comment:
nit, worth considering standardizing all doc string comments to `///` and `//!`, should save you two top and bottom lines as well. this seems to be the norm in other rust projects.

Reply:
I think this is super tedious and whatever "other rust projects do" doesn't convince me much unless there is a lint to enforce it :smile: 

I'll change it anyways, but I'm registering my protest :smiling_imp:

Reply:
haha, I would agree with the tedious claim if you didn't put a `*` at the beginning of every line ;)